lib/options.nix

```diff
@@ -403,5 +403,31 @@ rec {
         if cfg ? lazyLoad then lib.literalMD "`false` when lazy-loading is enabled." else true;
       example = false;
     };
+
+  mkMaybeUnpackagedPackage =
+    loc: pkgs: name: package:
+    if lib.isOption package then
+      package
+    else if package != null then
+      lib.mkPackageOption pkgs name {
+        nullable = true;
+        default = package;
+      }
+    else
+      (lib.mkOption {
+        type = types.nullOr types.package;
+        description = ''
+          Package to use for ${name}.
+          Nixpkgs does not include this package, and as such an external derivation or null must be provided.
+        '';
+        default = throw ''
```

Comment:
Looking at CI ([e.g.](https://buildbot.nix-community.org/#/builders/684/builds/3767)), I'm wondering if this fails because the module system does type-checking on all definitions, even those that are overridden?

I.e. when calling `opt.type.check` with the default definition, this will throw.

Reply:
At least one of the issues is related to the documentation generation

Reply:
Hm, even with `defaultText` we still have this in the [latest CI run](https://buildbot.nix-community.org/#/builders/1017/builds/3762):

```
error: Nixvim (plugins.lsp.servers.ada_ls.package): No package is known for ada_ls, to resolve this either:
  - install externally and set this option to `null`
  - or provide a derviation to install this package
```

Reply:
Ah, I bet it's coming from `url` attempting to get `meta.homepage`.

Reply:
And it only occurs in the mdbook documentation, not for the generation of man pages

Reply:
```
while evaluating the attribute 'package.default.meta.homepage'
```

:(

Reply:
> ```
> while evaluating the attribute 'package.default.meta.homepage'
> ```

See above: https://github.com/nix-community/nixvim/pull/2852#discussion_r1936381943